cmd/create.go

```diff
@@ -0,0 +1,80 @@
+package cmd
+
+import (
+	"fmt"
+	"kool-dev/kool/cmd/presets"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// KoolCreate holds handlers and functions to implement the preset command logic
+type KoolCreate struct {
+	DefaultKoolService
+	parser presets.Parser
+	KoolDocker
+	KoolPreset
+}
+
+func init() {
+	var (
+		create    = NewKoolCreate()
+		createCmd = NewCreateCommand(create)
+	)
+
+	rootCmd.AddCommand(createCmd)
+}
+
+// NewKoolCreate creates a new handler for create logic
+func NewKoolCreate() *KoolCreate {
+	return &KoolCreate{
+		*newDefaultKoolService(),
+		&presets.DefaultParser{Presets: presets.GetAll()},
+		*NewKoolDocker(),
+		*NewKoolPreset(),
+	}
+}
+
+// Execute runs the create logic with incoming arguments.
+func (c *KoolCreate) Execute(originalArgs []string) (err error) {
+	preset := originalArgs[0]
+	dir := originalArgs[1]
+
+	if !c.parser.Exists(preset) {
```

Comment:
maybe we only instantiate the `c.parser` here, which would make it lazy enough already.